app/views/contributions/show.html.erb

```diff
@@ -7,8 +7,13 @@
     <p class="subtitle"><%= contribution.description %></p>
     <% if contribution.matched? %>
       <i class="far fa-check-circle has-text-primary is-size-5 has-text-weight-semibold"><span class="ml-1">Claimed</span></i>
-    <% else %>
-      <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", method: :get %>
+    <% else  %>
+      <% if contribution.has_email? %>
+        <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", method: :get %>
+      <% else %>
+        <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", disabled: true, method: :get %>
```

Comment:
Very minor thought: you could simplify line 14 to something like:
```suggestion
        <button class="button is-primary" disabled="true">Claim This <%= contribution.type %></button>
```
because the disabled button can omit all the extra details needed (and rendered) by `button_to`.

Reply:
Great catch, thank you @exbinary!